packages/router/src/router-context.tsx

```diff
@@ -2,14 +2,17 @@ import React, { useReducer, createContext, useContext } from 'react'
 
 import { useAuth } from '@redwoodjs/auth'
 
-import { ParamType } from './internal'
+import type { ParamType } from 'src/internal'
+import { isRoute, PageType } from 'src/router'
+import { flattenAll } from 'src/util'
 
 const DEFAULT_PAGE_LOADING_DELAY = 1000 // milliseconds
 
 export interface RouterState {
   paramTypes?: Record<string, ParamType>
   pageLoadingDelay?: number
   useAuth: typeof useAuth
+  routes: Array<{ name: string; path: string; page: PageType }>
```

Comment:
```suggestion
  routes: Array<{ name?: string; path?: string; page?: PageType }>
```

This will make your warning further down go away.